tests/FSharpLint.Core.Tests/Rules/Conventions/FavourStaticEmptyFields.fs

```diff
@@ -0,0 +1,92 @@
+module FSharpLint.Core.Tests.Rules.Conventions.FavourStaticEmptyFields
+
+open NUnit.Framework
+open FSharpLint.Rules
+open System
+
+[<TestFixture>]
+type TestConventionsFavourStaticEmptyFields() =
+    inherit TestAstNodeRuleBase.TestAstNodeRuleBase(FavourStaticEmptyFields.rule)
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_1() =
+        this.Parse "let foo = \"\""
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_2() =
+        this.Parse "System.Console.WriteLine \"\""
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_3() =
+        this.Parse "let aList = []"
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_4() =
+        this.Parse "System.Console.WriteLine([].Length)"
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_5() =
+        this.Parse """
+let foo a =
+    if a = 0 then
+        "0"
+    else 
+        "" """
+
+        Assert.IsTrue this.ErrorsExist
+
+    [<Test>]
+    member this.FavourStaticEmptyFieldsShouldProduceError_6() =
+        this.Parse """
+match foo with
+| [] -> true
+| head::_ -> false"""
+
+        Assert.IsTrue this.ErrorsExist
```

Comment:
given that the F# snippet in the test FavourStaticEmptyFieldsShouldNotProduceError_7 doesn't compile, then this Assert should be the opposite: NoErrorsExist

Reply:
Done